src/mutableAtom/index.ts

```diff
@@ -30,31 +30,28 @@ export function makeMutableAtom<Value>(
 
   const storeAtom = atom<
     Store<Value>,
-    [ActionWithPayload<'setValueAsync', Value> | Action<'mount'>],
-    void
+    [ActionWithPayload<'setValue', Value> | Action<'getValue'>],
+    void | Value
   >(
-    (get, { setSelf }) => ({
-      proxyState: null,
-      getValue: () => get(valueAtom).value,
-      setValue: async (value: Value) => {
-        await defer()
-        setSelf({ type: 'setValueAsync', payload: value })
-      },
-    }),
+    (_get, { setSelf }) => {
+      const getValue = () => setSelf({ type: 'getValue' }) as Value
+      const store: Store<Value> = {
+        proxyState: createProxyState(getValue(), () => store),
```

Comment:
is `setSelf` called synchronous?

Reply:
Nice catch. I overlooked it.
Yeah, that's why I'd only support `mutableAtom`. New commit: f6e65fc